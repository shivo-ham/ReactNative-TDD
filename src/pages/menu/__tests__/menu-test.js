/**
 * @format
 */

import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Menu from '../';

describe('Menu', () => {
  const fakeNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };

  it('should render correctly', () => {
    //Check render view
    const rendered = renderer.create(<Menu navigation={fakeNavigation} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
