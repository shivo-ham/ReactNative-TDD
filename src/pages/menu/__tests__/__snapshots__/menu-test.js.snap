// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": undefined,
              "marginRight": undefined,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 5,
                "left": 5,
                "right": 5,
                "top": 5,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            >
              <Image
                isLoading={false}
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/menu.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 30,
                      "tintColor": undefined,
                      "width": 30,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              undefined,
              Object {
                "color": "#d56f51",
                "fontSize": 29,
                "fontWeight": "bold",
              },
            ]
          }
        >
          TOURX
        </Text>
        <View
          style={
            Object {
              "marginLeft": undefined,
              "marginRight": undefined,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 5,
                "left": 5,
                "right": 5,
                "top": 5,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            >
              <Image
                isLoading={false}
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/assets/user.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 50,
                      "tintColor": undefined,
                      "width": 50,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>,
]
`;
